### PR TITLE
Update zig to 0.11.0

### PR DESCRIPTION
**Pull request by 0x5a4**
I think it might be time to publish a release. 

The official zig package manager fetches dependencies as tarballs and at the moment you have to specify a url in the form of `https://github.com/mattyhall/tomlz/archive/<full-commit-hash>.tar.gz`.

**EDIT:** I removed the `zigmod` and `gyro` examples, as both seem to be no longer working and abandoned. The README has also been updated to reflect this and now correctly gives instructions for the official zig package manager.

### CODE REVIEW

**Comment by 0x5a4** [comment]
I've also finally installed nix, so you no longer have to update the lock-file 🥳

File: README.md [hunk 1]
```diff
@@ -32,46 +32,39 @@ decoding into a struct directly - including types that must be allocated like
...
-```
-$ gyro add mattyhall/tomlz
-$ gyro fetch
+In your `build.zig` file create and use the dependency
 ```
+pub fn build(b: *std.Build) void {
+    // ... setup ...
 
-An [example is available](https://github.com/mattyhall/tomlz/tree/main/examples/gyro).
+    const ziglua = b.dependency("tomlz", .{
```
Comment (mattyhall) [inline README.md:58]:
```suggestion
    const tomlz = b.dependency("tomlz", .{
```

File: README.md [hunk 2]
```diff
@@ -32,46 +32,39 @@ decoding into a struct directly - including types that must be allocated like
...
+        .target = target,
+        .optimize = optimize,
+    });
 
-### Submodule
-To add as a package a common approach is to clone this repository as a submodule
-in your own, and then add it as a package in `build.zig` like so:
+    // add the tomlz module 
+    exe.addModule("tomlz", ziglua.module("tomlz"));
```
Comment (mattyhall) [inline README.md:64]:
```suggestion
    exe.addModule("tomlz", tomlz.module("tomlz"));
```

File: flake.nix [hunk 3]
```diff
@@ -37,7 +37,7 @@
           rec {
             devShell = pkgs.mkShell {
               buildInputs = (with pkgs; [
-                zigpkgs.master-2023-07-02
+                zigpkgs."0.11.0"
```
Comment (mattyhall) [inline flake.nix:40]: We should continue to track master on the `main` branch. Once this change is in I'll branch `0.11.0` and we can fix the version there.

**Review by mattyhall — COMMENTED** [review]
Sorry it took so long to get around to this

**Review by mattyhall — APPROVED** [review]
Looks good. Thank you very much!

**Pull request merged**